source/manuals/programming-languages/java.html.md.erb

```diff
@@ -32,14 +30,16 @@ The IntelliJ "Optimize imports" command helps to sorting them alphabetically and
 
 ## Code checking
 
-Use of [Sonar](https://www.sonarqube.org/) for static analysis of code is encouraged.
+The use of static analysis is encouraged. Static analysis tools for Java include [SonarQube](https://www.sonarqube.org/), [Codacy](https://www.codacy.com/), [FindBugs](http://findbugs.sourceforge.net/) and [CheckStyle](http://checkstyle.sourceforge.net/). However, be aware that such tools can detect an overwhelming number of problems if applied to an existing project, which tends to result in their checks being ignored.
```

Comment:
The last sentence here puts me off following this advice. It's almost suggesting I shouldn't bother if I'm not setting up a brand new project.

Do we have any templates or examples of builds that have these set up? Is it possible to set up CI so that static analysis runs on the changes but not the whole project?

Reply:
I’m not aware of any templates or examples.

We could lose the subordinate clause from the last sentence, or perhaps add some advice of how to deal with the deluge (e.g. configure the tools to find fewer problems or have a prioritisation plan) and what to focus on (e.g. newer code — problems in code that’s been out there for ages can’t be that bad or someone would have complained already! — and more serious problems).

Reply:
I think I'd find the latter quite helpful, as someone who's very unfamiliar with the tooling.

Reply:
I have some thoughts on guidance for this but it’s not something we discussed in the meeting and I don’t feel qualified to write something unilaterally

Reply:
Ok, maybe it's best to address this as a separate PR then? Happy to discuss further if I can help at all, but I'm speaking more as a user of the guidance than someone with an opinion on how to do this.

Reply:
Yeah, perfect is the enemy of good and all that